tests/flows/validator_set_sig.go

```diff
@@ -0,0 +1,214 @@
+package flows
+
+import (
+	"context"
+	"math/big"
+
+	runner_sdk "github.com/ava-labs/avalanche-network-runner/client"
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	exampleerc20 "github.com/ava-labs/teleporter/abi-bindings/go/Mocks/ExampleERC20"
+	validatorsetsig "github.com/ava-labs/teleporter/abi-bindings/go/Utilities/ValidatorSetSig"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	. "github.com/onsi/gomega"
+)
+
+func ValidatorSetSig(network interfaces.LocalNetwork) {
+	// Deploy ValidatorSetSig contract to contract subnet to be validated from validator subnet
+	// Deploy a mock ERC20 contract to contract subnet
+	// Construct ValidatorSetSig message with mock ERC20 as the target contract
+	// Create a OffChainWarp Message using the ValidatorSetSig message to be signed by the validator subnet
+	// Confirm the event is emitted
+	// Retry the same message and confirm it fails
+	// Send a new message with incremented nonce and confirm it is accepted.
+	// Deploy a new mock ERC20 contract to the validator subnet
+	// Send a new message with the new ERC20 contract as the target contract and confirm it is accepted
+
+	contractSubnet, validatorSubnet := utils.GetTwoSubnets(network)
```

Comment:
Could just be me, but I find "contractSubnet" and "validatorSubnet" to be confusing names here. I think the intention was that "validatorSubnet" was that the `validatorSetSig contract deployed there is owned by that subnet itself.

Reply:
I agree. Original intention was to have `contractSubnet` have both contracts deployed on it, `validatorSetSig` and the `ExampleERC20` and the `validatorSubnet` is the subnet that signs the warp message to trigger actions. 

Given @cam-schultz 's feedback I added a test case where the signing chain is the same as the chain that contracts are deployed on which meant that I had to either:
:one: deploy an instance of the `validatorSetSig` contract on the `validatorSubnet` as well
:two: change how I was handling the warp messages and have the `contractSubnet` sign off-chain warp messages as well.

I opted for the former but kept the confusing naming. 
I'm changing to `subnetA` and `subnetB` now but happy to pivot to option :two: if that is preferred.

Reply:
I'm good with just `subnetA` and `subnetB` 👍  the test itself looks correct and provides good coverage as-is IMO